modules/warehouse/persistence/inventory_repository.go

```diff
@@ -17,121 +19,400 @@ func NewInventoryRepository() inventory.Repository {
 	return &GormInventoryRepository{}
 }
 
+func FormatLimitOffset(limit, offset int) string {
+	if limit > 0 && offset > 0 {
+		return fmt.Sprintf("LIMIT %d OFFSET %d", limit, offset)
+	} else if limit > 0 {
+		return fmt.Sprintf("LIMIT %d", limit)
+	} else if offset > 0 {
+		return fmt.Sprintf("OFFSET %d", offset)
+	}
+	return ""
+}
+
 func (g *GormInventoryRepository) GetPaginated(
 	ctx context.Context, params *inventory.FindParams,
 ) ([]*inventory.Check, error) {
-	tx, ok := composables.UseTx(ctx)
-	if !ok {
-		return nil, composables.ErrNoTx
+	pool, err := composables.UsePool(ctx)
+	if err != nil {
+		return nil, err
 	}
-	tx = tx.Limit(params.Limit).Offset(params.Offset)
-	if params.Query != "" && params.Field != "" {
-		tx = tx.Where(fmt.Sprintf("%s::varchar ILIKE ?", params.Field), "%"+params.Query+"%")
+	where, args := []string{"1 = 1"}, []interface{}{}
+
+	if params.Status != "" {
+		where, args = append(where, fmt.Sprintf("ic.status = $%d", len(args)+1)), append(args, params.Status)
 	}
 
-	if params.CreatedAt.To != "" && params.CreatedAt.From != "" {
-		tx = tx.Where("created_at BETWEEN ? and ?", params.CreatedAt.From, params.CreatedAt.To)
+	if params.Type != "" {
+		where, args = append(where, fmt.Sprintf("ic.type = $%d", len(args)+1)), append(args, params.Type)
 	}
 
-	if params.Status != "" {
-		tx = tx.Where("status = ?", params.Status)
+	if params.CreatedAt.To != "" && params.CreatedAt.From != "" {
+		where, args = append(where, fmt.Sprintf("ic.created_at BETWEEN $%d and $%d", len(args)+1, len(args)+2)), append(args, params.CreatedAt.From, params.CreatedAt.To)
 	}
 
-	if params.Type != "" {
-		tx = tx.Where("type = ?", params.Type)
+	if params.Query != "" && params.Field != "" {
+		where, args = append(where, fmt.Sprintf("ic.%s::VARCHAR ILIKE $%d", params.Field, len(args)+1)), append(args, "%"+params.Query+"%")
 	}
 
-	tx, err := helpers.ApplySort(tx, params.SortBy)
+	rows, err := pool.Query(ctx, `
```

Comment:
This query could become a problem if you are using graphql endpoints, and you want `inventory_checks->created_by->avatar`, for example. Maybe it's better to drop joins, and use individual selects?

```go
if check.CreatedBy, err = userRepo.FindUserByID(ctx, check.CreatedByID) {

}
```

Reply:
Yeah, why not. Most ORMs just don't do joins and do separate Selects. If it starts hitting performance at some point we'll come back to this